File: app/src/main/java/co/medellinandroid/spotifyartists/model/SpotifyArtistsItems.java
package co.medellinandroid.spotifyartists.model;

import com.squareup.moshi.Json;
import java.util.ArrayList;
import java.util.List;

public class SpotifyArtistsItems {

  @Json(name = "id") private String id;
  @Json(name = "name") private String name;
  @Json(name = "uri") private String uri;
  @Json(name = "images") private List<SpotifyArtistImage> images = new ArrayList<>();

  public SpotifyArtistsItems() {
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getId() {
    return id;
  }

  public void setId(String id) {
    this.id = id;
  }

  public String getUri() {
    return uri;
  }

  public void setUri(String uri) {
    this.uri = uri;
  }

  public List<SpotifyArtistImage> getImages() {
    return images;
  }

  public void setImages(List<SpotifyArtistImage> images) {
    this.images = images;
  }

  public static class Builder {

    private SpotifyArtistsItems spotifyArtist = new SpotifyArtistsItems();

    public Builder withName(String name) {
      spotifyArtist.setName(name);
      return this;
    }

    public SpotifyArtistsItems build() {
      return spotifyArtist;
    }
  }
}

File: app/src/main/java/co/medellinandroid/spotifyartists/model/SpotifyArtistsItems.kt
package co.medellinandroid.spotifyartists.model

import com.squareup.moshi.Json
import java.util.ArrayList

class SpotifyArtistsItems {

  @Json(name = "id")
  var id: String? = null
  @Json(name = "name")
  var name: String? = null
  @Json(name = "uri")
  var uri: String? = null
  @Json(name = "images")
  var images: List<SpotifyArtistImage> = ArrayList()

  class Builder {
    private val spotifyArtist = SpotifyArtistsItems()

    fun withName(name: String): Builder {
      spotifyArtist.name = name
      return this
    }

    fun build(): SpotifyArtistsItems = spotifyArtist
  }
}
